src/main/java/edu/umich/tl/ApiCallHandler.java

```diff
@@ -112,7 +187,7 @@ private HttpResponse apiDirectCanvas(String url) {
 		return httpResponse;
```

Comment:
Nice catch!. I have added a null check now. I did not realize it could send a null.